src/main/helpers.ts

```diff
@@ -129,7 +125,7 @@ export function onNavigation({ urlTarget, currentUrl, preventDefault, createNewW
   }
 
   const sameHost = targetUrl.hostname === url.hostname;
-  const sameApp = sameHost && targetUrl.pathname.startsWith(url.pathname);
+  const sameApp = sameHost && targetUrl.pathname.split('/')[1] === url.pathname.split('/')[1];
```

Comment:
I don't think this is what we want either, but I also think this variable can just be removed here and from the check below.